internal/catalogmetadata/types_test.go

```diff
@@ -229,3 +229,55 @@ func TestBundleHasDeprecation(t *testing.T) {
 		})
 	}
 }
+
+func TestBundleIsDeprecated(t *testing.T) {
+	for _, tt := range []struct {
+		name       string
+		bundle     *catalogmetadata.Bundle
+		deprecated bool
+	}{
+		{
+			name: "has package and channel deprecations, not deprecated",
+			bundle: &catalogmetadata.Bundle{
+				Deprecations: []declcfg.DeprecationEntry{
+					{
+						Reference: declcfg.PackageScopedReference{
+							Schema: "olm.package",
+						},
+					},
+					{
+						Reference: declcfg.PackageScopedReference{
+							Schema: "olm.channel",
+							Name:   "foo",
+						},
+					},
+				},
+			},
+		},
+		{
+			name:   "has no deprecation entries, not deprecated",
+			bundle: &catalogmetadata.Bundle{},
```

Comment:
Possibly a silly question: Does this `catalogmetadata.Bundle{}` need to have the package and channel deprecations listed for them to show up in the ClusterExtension deprecation status conditions, or is that handled in a separate data structure?

If it is handled in a separate data strucutre, could we change `catalogmetadata.Bundle` to contain something like:

```go
package catalogmetadata

type Bundle struct {
	...
	deprecated bool
}


```

It seems unnecessary to store a list of `Deprecations` where there will always be 0 or 1 and the schema/name will always be `olm.bundle` and `bundle.Name`.

Reply:
I think this question is related to https://github.com/operator-framework/operator-controller/pull/577#discussion_r1452452552